client/src/plugins/camunda-plugin/deployment-tool/__tests__/DeploymentConfigOverlaySpec.js

```diff
@@ -138,39 +137,38 @@ describe('<DeploymentConfigOverlay>', () => {
         }
       };
 
+      const validator = new MockValidator({
+        validateConnection: () => new Promise((resolve, err) => {
+          resolve({
+            code: GenericApiErrors.UNAUTHORIZED
+          });
+        })
+      });
+
       const {
         wrapper,
         instance
       } = createOverlay({
         anchor,
-        configuration
+        configuration,
+        validator
       }, mount);
 
       // when
-      act(() => {
-        instance.setState({ isAuthNeeded: true });
-      });
-
-      instance.isOnBeforeSubmit = true;
-
-      wrapper.update();
+      setTimeout(() => {
```

Comment:
I understand that the current test suite for the deployment tool is in bad shape, but I'd rather expect us to properly make use of `act` and `waitFor` instead of introducing more `setTimeout` calls. Let's not make the tests depend on the time.

Reply:
`waitFor` is not yet available in this library as we do need have `@testing-library/react` as a (dev) dependency at the moment. Solely using `act` I was not able get it working. Also, I am not sure about the Formik workaround with the `isOnBeforeSubmit` flag.

Reply:
I managed to get this working now, but I had to introduce a new dependency `@testing-library/react` which you mentioned.